server/src/main/java/org/opensearch/extensions/ExtensionDenpendency.java

```diff
@@ -0,0 +1,56 @@
+package org.opensearch.extensions;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import org.opensearch.Version;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+
+public class ExtensionDenpendency implements Writeable {
+    public String uniqueId;
+    public Version version;
+
+    public ExtensionDenpendency(String uniqueId, Version version){
+        this.uniqueId = uniqueId;
+        this.version = version;
+        List<ExtensionDenpendency> dependencies = Collections.<ExtensionDenpendency>emptyList();
```

Comment:
No need to add a generic type for `Collections.emptyList()`. It is inferred from the type definition.